Update whosnext.py

ThomasHabing opened the pull request:
The selection was weighted but was not random. Now, it is still weighted but random. Added Thomas October 8th 2024

ThomasHabing committed:
Update whosnext.py

The selection was weighted but was not random. Now, it is still weighted but random.
Added Thomas October 8th 2024

moorepants commented on whosnext.py:
```diff
@@ -167,10 +170,12 @@
+    remaining_members = [member for member in current_members if member not in choice]
+    remaining_weights = [weights[member] for member in remaining_members]
+    chosen = random.choices(remaining_members, weights=remaining_weights, k=1)[0]
+    choice.add(chosen)
```
What exactly are you fixing here? What was wrong with the prior code?

ThomasHabing replied:
The previous code was not a random weighted selection, just a weighted selection. It would always select the same people.

moorepants committed:
Cleaned up Kenneth's method.

- The only way to have 0 weight: MC or presented last meeting.
- Weights are days since last presentation up to max of 180 (6 months *
  30 days).
- Added missing presentations.

moorepants commented:
@ThomasHabing I made these changes:

- The only way to have 0 weight: MC or presented last meeting.
- Weights are days since last presentation up to max of 180 (6 months *
  30 days).
- Added missing presentations.

I've tested it many times and I print the weights to the screen before making the two choices. I will merge. Please test it before using it next time to see if you agree on it.  You can disable the roulette with `ROULETTE` variable for fast testing.
